cudax/include/cuda/experimental/__memory_resource/any_resource.cuh

```diff
@@ -95,18 +233,34 @@ public:
           nullptr, &_CUDA_VMR::__alloc_vtable<_Alloc_type, _CUDA_VMR::_WrapperType::_Owning, __resource_t>)
       , __vtable(__vtable::template _Create<__resource_t>())
   {
-    if constexpr (_CUDA_VMR::_IsSmall<__resource_t>())
-    {
-      ::new (static_cast<void*>(this->__object.__buf_)) __resource_t(_CUDA_VSTD::forward<_Resource>(__res));
-    }
-    else
-    {
-      this->__object.__ptr_ = new __resource_t(_CUDA_VSTD::forward<_Resource>(__res));
-    }
+    static_assert(_CUDA_VMR::_IsSmall<__resource_t>());
+    ::new (static_cast<void*>(this->__object.__buf_)) __resource_t(_CUDA_VSTD::forward<_Resource>(__res));
+  }
+
+  //! @brief Constructs a \c basic_any_resource wrapping an object of type \c _Resource that
+  //! is constructed from \c __args... . \c _Resource must satisfy the  \c resource or \c async_resource
+  //! concept, and it must provide all properties in \c _Properties... .
+  //!
+  //! @param __args The arguments used to construct the instance of \c _Resource this \c basic_any_resource
+  //! object wraps.
+  //!
+  //! @note If \c _Resource is copyable, nothrow-movable, and sufficiently small (`<= 3*sizeof(void*)`),
+  //! it will be stored directly within the \c basic_any_resource. Otherwise, it will be shared between all
+  //! copies of the \c basic_any_resource.
```

Comment:
Critical: I believe that is a bad default. We should either always ref count or never. Case in point our `async_memory_resource` is a pooled resource but only stores a single pointer.

Reply:
That's actually what happens now, the comment is out of date. I'll fix it.